src/hb-font.cc

```diff
@@ -1854,10 +1870,25 @@ hb_font_set_variations (hb_font_t *font,
   if (unlikely (coords_length && !normalized))
     return;
 
+  float *design_coords = coords_length ? (float *) calloc (coords_length, sizeof (float)) : nullptr;
+  if (unlikely (coords_length && !design_coords))
+  {
+    free (normalized);
+    return;
+  }
+
+  for (unsigned int i = 0; i < variations_length; i++)
+  {
+    hb_ot_var_axis_info_t info;
+    if (hb_ot_var_find_axis_info (font->face, variations[i].tag, &info) &&
+	info.axis_index < coords_length)
+      design_coords[info.axis_index] = variations[i].value;
+  }
```

Comment:
Rewrite this function to populate design coords completely, then normalize them...

Reply:
don't know what "completely" means here but have merged normalize to the code no extra axis fetching.

Reply:
Too much of duplication as it used to call `hb_ot_var_normalize_variations` which does another `hb_ot_var_find_axis_info` and if I just normalize every axis it makes `tests/GPOS-5.tests` fail, can upload on a separate PR.